Index: feel++-0.102.0~git+20170511/cmake/modules/feelpp.dependencies.cmake
===================================================================
--- feel++-0.102.0~git+20170511.orig/cmake/modules/feelpp.dependencies.cmake
+++ feel++-0.102.0~git+20170511/cmake/modules/feelpp.dependencies.cmake
@@ -53,7 +53,8 @@ IF( ("${CMAKE_CXX_COMPILER_ID}" MATCHES
     ("${CMAKE_CXX_COMPILER_ID}" MATCHES "AppleClang") OR
     ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Intel") )
 
-  set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++${FEELPP_STD_CPP}" )
+  set (CMAKE_CXX_STANDARD ${FEELPP_STD_CPP})
+  #set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++${FEELPP_STD_CPP}" )
 
   if ( NOT ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Intel") )
     set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ftemplate-depth=1024" )
