From be8e5cf28ae1411ca14ee730db77ec4cac2c051e Mon Sep 17 00:00:00 2001
From: Vincent Chabannes <vincent.chabannes@feelpp.org>
Date: Thu, 18 May 2017 10:07:08 +0200
Subject: [PATCH 4/4] add feelpp subdir

---
 cmake/machines/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/machines/CMakeLists.txt b/cmake/machines/CMakeLists.txt
index bf7ce7122..1f5ad52eb 100644
--- a/cmake/machines/CMakeLists.txt
+++ b/cmake/machines/CMakeLists.txt
@@ -24,4 +24,4 @@
 #
 FILE(GLOB files "${CMAKE_CURRENT_SOURCE_DIR}/*.cmake")
 
-INSTALL(FILES ${files} DESTINATION share/feel/cmake/machines COMPONENT Devel)
+INSTALL(FILES ${files} DESTINATION share/feelpp/feel/cmake/machines COMPONENT Devel)
-- 
2.11.0

